src/h264/h264.h

```diff
@@ -180,8 +226,328 @@ class H264SeqParameterSet
+			cpb_size_scale = r.Get(4);
+			for (uint32_t SchedSelIdx = 0; SchedSelIdx <= cpb_cnt_minus1; SchedSelIdx++)
+			{
+				bit_rate_value_minus1.push_back(r.GetExpGolomb());
+				cpb_size_value_minus1.push_back(r.GetExpGolomb());
+				cbr_flag.push_back(r.Get(1));
+			}
+			initial_cpb_removal_delay_length_minus1 = r.Get(5);
+			cpb_removal_delay_length_minus1 = r.Get(5);
+			dpb_output_delay_length_minus1 = r.Get(5);
+			time_offset_length = r.Get(5);
+			return true;
+		}
+		void DumpHrdParams() const 
+		{
+			Debug("\t\t\tcpb_cnt_minus1=%u\n",	cpb_cnt_minus1);
+			Debug("\t\t\tbit_rate_scale=%u\n",	bit_rate_scale);
+			Debug("\t\t\tcpb_size_scale=%u\n",	cpb_size_scale);
+			for (uint32_t SchedSelIdx = 0; SchedSelIdx <= cpb_cnt_minus1; SchedSelIdx++)
+			{
+				Debug("\t\t\tbit_rate_value_minus1[%u]=%u\n",	SchedSelIdx, cpb_size_scale);
+				Debug("\t\t\tcpb_size_value_minus1[%u]=%u\n",	SchedSelIdx, cpb_size_value_minus1);
+				Debug("\t\t\tcbr_flag[%u]=%u\n",	SchedSelIdx, cbr_flag);
+			}
+			Debug("\t\t\tinitial_cpb_removal_delay_length_minus1=%u\n",	initial_cpb_removal_delay_length_minus1);
+			Debug("\t\t\tcpb_removal_delay_length_minus1=%u\n",	cpb_removal_delay_length_minus1);
+			Debug("\t\t\tdpb_output_delay_length_minus1=%u\n",	dpb_output_delay_length_minus1);
+			Debug("\t\t\ttime_offset_length=%u\n",	time_offset_length);
+		}
+	};
+
+	struct VuiParameters
+	{
+		uint32_t      aspect_ratio_info_present_flag;
+		uint32_t      aspect_ratio_idc;
+		std::optional<uint32_t>      sar_width = {};
+		std::optional<uint32_t>      sar_height = {};
+		uint32_t      overscan_info_present_flag = 0;
+		uint32_t      overscan_appropriate_flag = 0;
+		uint32_t  	  video_signal_type_present_flag = 0;
+		uint32_t 	  video_format = 0;
+		uint32_t 	  video_full_range_flag = 0;
+		uint32_t  	  colour_description_present_flag = 0;
+		uint32_t 	  colour_primaries = 0;
+		uint32_t 	  transfer_characteristics = 0;
+		uint32_t 	  matrix_coefficients = 0;
+		uint32_t 	  chroma_loc_info_present_flag = 0;
+		uint32_t 	  chroma_sample_loc_type_top_field = 0;
+		uint32_t	  chroma_sample_loc_type_bottom_field = 0;
+		uint32_t 	  timing_info_present_flag = 0;
+		uint32_t 	  num_units_in_tick = 0;
+		uint32_t 	  time_scale = 0;
+		uint32_t 	  fixed_frame_rate_flag = 0;
+		uint32_t      nal_hrd_parameters_present_flag = 0;
+		HrdParameters nal_hrd_parameters;
+		uint32_t      vcl_hrd_parameters_present_flag = 0;
+		HrdParameters vcl_hrd_parameters;
+		uint32_t      low_delay_hrd_flag = 0;
+		uint32_t 	  pic_struct_present_flag = 0;
+		uint32_t 	  bitstream_restriction_flag = 0;
+		uint32_t 	  motion_vectors_over_pic_boundaries_flag = 0;
+		uint32_t 	  max_bytes_per_pic_denom = 0;
+		uint32_t 	  max_bits_per_mb_denom = 0;
+		uint32_t 	  log2_max_mv_length_horizontal = 0;
+		uint32_t 	  log2_max_mv_length_vertical = 0;
+		uint32_t 	  max_num_reorder_frames = 0;
+		uint32_t 	  max_dec_frame_buffering = 0;
+		bool DecodeVuiParameters(RbspBitReader &r)
```

Comment:
keep naming consistency
```suggestion
		bool Decode(RbspBitReader &r)
```